### ISSUE TITLE
Firebase Suppport?

### ISSUE THREAD

**Issue by Kansattica** [Redth/PushSharp#711]
Google recently released their new notification engine, [Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/). Will PushSharp support it in the near future?

**Comment by AdhamAwadhi** [comment]
Will support?

**Comment by siegerx3** [comment]
Theoreticall it's only changing the url and it should work as before, correct?

**Comment by siegerx3** [comment]
Actually, just call 
`OverrideUrl()`
on the GcmConfiguration. And use this url "https://fcm.googleapis.com/fcm/send".

Worked for me :) # #

**Comment by mcvasquez** [comment]
@siegerx3  It doesn't worked for me. Could you post an example, please?

**Comment by siegerx3** [comment]
`var gcmConfig = new GcmConfiguration("APIKEY");
  gcmConfig.GcmUrl = "https://fcm.googleapis.com/fcm/send";`

This is what i'm doing and what is working for me.
I can't tell you tho if your problem comes from somewhere else.

**Comment by mcvasquez** [comment]
@siegerx3 I was using `config.OverrideUrl("https://fcm.googleapis.com/fcm/send");` as you specified before. It works now using `gcmConfig.GcmUrl = "https://fcm.googleapis.com/fcm/send";`. Thanks.

**Comment by siegerx3** [comment]
@mcvasquez 
Yeah sorry about that, i first saw the method and then i saw that you can simply use the property.
Glad it works now tho :)

**Comment by hamzauzumcu** [comment]
@siegerx3 Thanks bro you save my day

**Comment by menasheu** [comment]
If I use the Firebase console, I can send a notification to my device using the identifiers of the application (senderID, Server Key). If I use Pushsharp with the above changes in the URL and using the same application identifiers, my callback says that the notification was successful but my device does not receive it. Any ideas?

**Comment by siegerx3** [comment]
@menasheu 
Can you show the code how you build your notification?
And to what device are you sending? IOS or Android?

**Comment by menasheu** [comment]
Sorry but I forgot to mention that it is now working fine. I use the code and directives supplied by Google on the client side for Android. Haven't tried yet iOS. For the server side, the above change on the URL just did it. Thanks anyway for inquiring.

**Comment by TrezeP** [comment]
Hi everyone,
I am using pussharp with GCM, and trying to migrate do FCM.
Do i need to change it in the app?

Just now i am having some strange behaviors, like i send 1 push and the cellphone receive it 3 times... any ideas?

I followed the example of "sieger3x" 

Thanks

**Comment by naveedahmed1** [comment]
Any update on this?

**Comment by siegerx3** [comment]
@naveedahmed1 
Do you have any particular problem with using FCM and PushSharp?

**Comment by thgreasi** [comment]
@siegerx3 I've been struggling with this but can't make the message to reach the device.
The `OnNotificationSucceeded` is invoked and no error shows up.
The target registrationID seems to be correct, since when using it on the web message composer @ `console.firebase.google.com` the message shows up.

**Comment by thgreasi** [comment]
Ok, so it was a mobile dev issue not having a proper Notification handler... So embarrassed, sorry for the extra noise.

**Comment by siegerx3** [comment]
@thgreasi 
No worries :)

**Comment by ElNinjaGaiden** [comment]
Hello everyone. I'm trying to use PushSharp + Firebase and it seems it works fine but I'm having a hard time trying to try to send the push to a specific topic instead to a single or a group of specific device registrations.

Any idea how can this be done??
Thanks in advance.
